### PR TITLE
util/types: add test for datum eval function

### PR DESCRIPTION
**Pull request by tiancaiamao**
Improve file datum_test.go test coverage to 93.6%

### CODE REVIEW

**Comment by shenli** [comment]
LGTM
@XuHuaiyu PTAL

File: util/types/datum_test.go [hunk 1]
```diff
@@ -366,3 +368,205 @@ func (ts *testDatumSuite) TestToBytes(c *C) {
...
+		minus  Datum
+		hasErr bool
+	}{
+		{NewIntDatum(72), NewIntDatum(28), NewIntDatum(100), NewIntDatum(44), false},
+		{NewIntDatum(72), NewUintDatum(28), NewIntDatum(100), NewIntDatum(44), false},
+		{NewUintDatum(72), NewUintDatum(28), NewUintDatum(100), NewUintDatum(44), false},
+		{NewUintDatum(72), NewIntDatum(28), NewUintDatum(100), NewUintDatum(44), false},
+		{NewFloat64Datum(72.0), NewFloat64Datum(28.0), NewFloat64Datum(100.0), NewFloat64Datum(44.0), false},
+		{NewDecimalDatum(NewDecFromStringForTest("72.5")), NewDecimalDatum(NewDecFromInt(3)), NewDecimalDatum(NewDecFromStringForTest("75.5")), NewDecimalDatum(NewDecFromStringForTest("69.5")), false},
+		{NewIntDatum(72), NewFloat64Datum(42), Datum{}, Datum{}, true},
```
Comment (XuHuaiyu) [inline util/types/datum_test.go]: what error do we get here?

Comment (tiancaiamao) [inline util/types/datum_test.go]: ```Invalid operation: %v %v %v (mismatched types %T and %T)```
`ComputePlus` and `ComputeMinus` don't handle different types except int/uint, and don't cast input to decimal.

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request merged**